addons/web_editor/static/src/js/editor/odoo-editor/test/spec/copyPaste.test.js

```diff
@@ -2028,6 +2028,66 @@ describe('Paste', () => {
             });
         });
     });
+    describe('Pasting h1', () => {
+        describe('range collapsed', async () => {
+            it('should paste h1 when pasting in empty p', async () => {
+                await testEditor(BasicEditor, {
+                    contentBefore: '<p>[]<br></p>',
+                    stepFunction: async editor => {
+                        await pasteHtml(editor, '<h1>a</h1><h1>b</h1><h1>c</h1>');
+                    },
+                    contentAfter: '<h1>a</h1><h1>b</h1><h1>c</h1><p>[]<br></p>',
```

Comment:
I'm really confused about the spec of this commit tbh. I would expect either (1)
```html
<p>a</p><h1>b</h1><h1>c[]</h1>
```
(which I believe is the case now), (2)
```html
<h1>a</h1><h1>b</h1><h1>c[]</h1>
```
or (3)
```html
<p><br></p><h1>a</h1><h1>b</h1><h1>c[]</h1>
```
but what you have here seems like we're pasting _before_ the selection. I understand it can be seen as "we add a paragraph after pasting in the (2) way, so that we can write in normal paragraph mode again" but I still find that weird.

@dmo-odoo

Reply:
(1) is wrong and is the purpose of the original task that was cancelled. (3) will definitely feel buggy when using it IMO. (2) is the way to go. From what I can see it is what has been done in the current PR.

Reply:
I agree that (2) is the way to go. The weird thing here is this test shows (2) but with an added `<p><br></p>` at the end.

Reply:
I have fixed that.